Helper/PCAHelper.cs

```diff
@@ -0,0 +1,220 @@
+        private string[] _scopes;
+
+        /// <summary>
+        /// Initializes the PCAHelpr or its derived class as per the generics and PublicClientApplicationBuilder with the given parameters
+        /// PublicClientApplicationBuilder can be customized after this method prior to accessing PublicClientApplication.
+        /// By default it is singleton pattern with option to force creation. 
+        /// </summary>
+        /// <typeparam name="T">Any class that is inherited from PCAHelper</typeparam>
+        /// <param name="clientId">Client id of your application</param>
+        /// <param name="scopes">The desired scope</param>
+        /// <param name="specialRedirectUri">If you are using recommended pattern fo rredirect Uri, this is optional</param>
+        /// <param name="forceCreate">Creates a new instance irrespective of the existance of the previous instance</param>
+        /// <returns>Instance of class inherited from PCAHelper</returns>
+        public static IPCAHelper Init<T>(string clientId, string[] scopes, string specialRedirectUri = null, bool forceCreate = false) 
+                where T : PCAHelper, new()
+        {
+            if (Instance == null || forceCreate)
+            {
+                var pcaHelper = new T();
+                pcaHelper._scopes = scopes;
+                pcaHelper.PCABuilder = PublicClientApplicationBuilder.Create(clientId)
+                                                                    .WithRedirectUri(specialRedirectUri ?? $"msal{clientId}://auth")
+                                                                    .WithIosKeychainSecurityGroup("com.microsoft.adalcache");
+                Instance = pcaHelper;
+            }
+            return Instance;
+        }
+
+        /// <summary>
+        /// This encapuslates the common pattern to acquire token i.e. attempt AcquireTokenSilent and if that throws MsalUiRequiredException attempt acquire token interactively.
+        /// Interactive attempt is optional.
+        /// It provides optional delegates to customize behavior.
+        /// </summary>
+        /// <param name="doSilent">Determines whether to execute AcquireTokenSilent</param>
+        /// <param name="doInteractive">Determines whether to execute AcquireTokenInteractive. By detault, UI interaction takes place if silent action fails.</param>
+        /// <param name="preferredAccount">Function that determines the account to be used. The default is first. (optional)</param>
+        /// <param name="customizeSilent">This is a delegate to optionally customize AcquireTokenSilentParameterBuilder.</param>
+        /// <param name="customizeInteractive">This is a delegate to optionally customize AcquireTokenInteractiveParameterBuilder.</param>
+        /// <returns>Authenitcation result</returns>
+        public async Task<AuthenticationResult> EnsureAuthenticatedAsync(
+                                                                bool doSilent = true,

```

Comment:
I think we should always do silent, even with a force refresh.

Reply:
I think we should provide flexibility to the user.

Reply:
Removed the flags